src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -301,7 +314,32 @@ public IDisposable BeginScope<TState>(TState state)
                 throw new ArgumentNullException(nameof(state));
             }
 
-            return DictionaryLoggerScope.Push(state);
+            var stateValues = state as IDictionary<string, object>;
+            if (stateValues == null)
+            {
+                // There's nothing we can do without a dictionary.
+                return null;
+            }
+
+            string functionName = stateValues.GetValueOrDefault<string>(ScopeKeys.FunctionName);
+            string functionInvocationId = stateValues.GetValueOrDefault<string>(ScopeKeys.FunctionInvocationId);
+            string eventName = stateValues.GetValueOrDefault<string>(ScopeKeys.Event);
+
+            // If we have the invocation id, function name, and event, we know it's a new function. That means
+            // that we want to start a new operation and let App Insights track it for us.
+            if (!string.IsNullOrEmpty(functionName) &&
+                !string.IsNullOrEmpty(functionInvocationId) &&
+                eventName == LogConstants.FunctionStartEvent)
+            {
+                RequestTelemetry request = new RequestTelemetry();
```

Comment:
Instead of the conditional check on state, I think I'd prefer to have a separate `BeginInvocationScope` method that adds the function invocation state, starts the operation, and delegates the rest to normal `BeginScope`. Is that possible?